test/performance/odp_stash_perf.c

```diff
@@ -0,0 +1,483 @@
+		}
+	}
+
+	if (test_options->num_stash > MAX_STASHS) {
+		ODPH_ERR("Too many stashes %u. Test maximum %u.\n",
+			 test_options->num_stash, MAX_STASHS);
+		return -1;
+	}
+
+	return ret;
+}
+
+static int create_stashes(test_global_t *global)
+{
+	uint32_t i;
+	uint32_t tmp = 0;
+	test_options_t *test_options = &global->options;
+
+	uint32_t num_stash = test_options->num_stash;
+	uint32_t num_round = test_options->num_round;
+	odp_stash_t *stash = global->stash;
+	odp_stash_capability_t stash_capa;
+
+	printf("\nTesting %s stashs\n",
+	       test_options->strict == 0 ? "NORMAL" : "STRICT_SIZE");
+	printf("  num rounds           %u\n", num_round);
+	printf("  num stashs           %u\n", num_stash);
+	printf("  stash size           %u\n", test_options->stash_size);
+	printf("  max burst size       %u\n", test_options->max_burst);
+
+	if (odp_stash_capability(&stash_capa, ODP_STASH_TYPE_DEFAULT)) {
+		printf("stash capability failed\n");
+		return -1;
+	}
+
+	if (test_options->stash_size > stash_capa.max_num_obj) {
+		ODPH_ERR("Max stash size supported %lu\n", stash_capa.max_num_obj);
+		return -1;
+	}
+
+	if (test_options->num_stash > stash_capa.max_stashes) {
+		ODPH_ERR("Max stash supported %u\n", stash_capa.max_stashes);
+		return -1;
+	}
+
+	for (i = 0; i < num_stash; i++) {
+		odp_stash_param_t stash_param;
+
+		odp_stash_param_init(&stash_param);
+		stash_param.num_obj    = test_options->stash_size;
+		stash_param.obj_size   = sizeof(uint32_t);
+		stash_param.strict_size = test_options->strict;
+
+		stash[i] = odp_stash_create("test_stash_u32", &stash_param);
+		if (stash[i] == ODP_STASH_INVALID) {
+			ODPH_ERR("Stash create failed\n");
+			return -1;
+		}
+
+		if (odp_stash_put_u32(stash[i], &tmp, test_options->max_burst) < 0) {
+			ODPH_ERR("Error: Stash put failed %u\n", i);
+			return -1;
+		}
+}
+
+	return 0;
+}
+
+static int destroy_stashes(test_global_t *global)
+{
+	odp_stash_t *stash = global->stash;
+	test_options_t *test_options = &global->options;
+	uint32_t num_stash = test_options->num_stash;
+
+	for (uint32_t i = 0; i < num_stash; i++) {
+		if (odp_stash_destroy(stash[i])) {
+			ODPH_ERR("Stash destroy failed\n");
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+static int run_test(void *arg)
+{
+	uint64_t c1, c2, cycles, nsec;
+	odp_time_t t1, t2;
+	uint32_t rounds;
+	int num_stored;
+	test_stat_t *stat;
+	test_global_t *global = arg;
+	test_options_t *test_options = &global->options;
+	odp_stash_t stash;
+	uint32_t tmp[32] = {0};
+	uint64_t num_retry = 0;
+	uint64_t ops = 0;
+	uint32_t num_stash = test_options->num_stash;
+	uint32_t num_round = test_options->num_round;
+	int thr = odp_thread_id();
+	int ret = 0;
+	uint32_t i = 0;
+	uint32_t max_burst = test_options->max_burst;
+
+	stat = &global->stat[thr];
+
+	/* Start all workers at the same time */
+	odp_barrier_wait(&global->barrier);
+
+	t1 = odp_time_local();
+	c1 = odp_cpu_cycles();
+
+	for (rounds = 0; rounds < num_round; rounds++) {
+		stash = global->stash[i++];
```

Comment:
Value of `i` is not correct for the couple error messages below.

Reply:
Removed the `i` printing to make logic simple.